ovs/lib/albaupdate.py

```diff
@@ -0,0 +1,432 @@
+
+                            information[key]['services_post_update'].update(framework_arakoons)
+                            if arakoon_ovs_info['down'] is True and ['ovsdb', None] not in information[key]['downtime']:
+                                information[key]['downtime'].append(['ovsdb', None])
+                        else:
+                            for down in arakoon_downtime:
+                                if down not in information[key]['downtime']:
+                                    information[key]['downtime'].append(down)
+                            information[key]['services_post_update'].update(arakoon_services)
+
+            for alba_node in AlbaNodeList.get_albanodes():
+                for package_name, package_info in alba_node.package_information.get(key, {}).iteritems():
+                    if package_name not in AlbaUpdateController.sdm_packages:
+                        continue  # Only gather information for the SDM packages
+
+                    information[key]['services_post_update'].update(package_info.pop('services_to_restart'))
+                    if package_name not in information[key]['packages']:
+                        information[key]['packages'][package_name] = {}
+                    information[key]['packages'][package_name].update(package_info)
+        return information
+
+    @staticmethod
+    @add_hooks('update', 'package_install_multi')
+    def package_install_alba_plugin(client, package_names):
+        """
+        Update the Alba plugin packages
+        :param client: Client on which to execute update the packages
+        :type client: SSHClient
+        :param package_names: Packages to install
+        :type package_names: list
+        :return: None
+        """
+        for package_name in package_names:
+            if package_name in AlbaUpdateController.alba_plugin_packages:
+                PackageManager.install(package_name=package_name, client=client)
+
+    @staticmethod
+    @add_hooks('update', 'package_install_single')
+    def package_install_sdm(package_names):
+        """
+        Update the SDM packages
+        :param package_names: Packages to install
+        :type package_names: list
+        :return: None
+        """
+        if set(package_names).intersection(AlbaUpdateController.sdm_packages):
+            for alba_node in AlbaNodeList.get_albanodes():
+                AlbaUpdateController._logger.debug('{0}: Updating SDM'.format(alba_node.ip))
+                try:
+                    result = alba_node.client.execute_update(status=None)
+                    # Backwards compatibility
+                    if result and 'status' in result and result['status'] != 'done':
+                        counter = 0
+                        max_counter = 12
+                        while counter < max_counter:
+                            status = alba_node.client.execute_update(status=None).get('status')
```

Comment:
Is this status looping still required? I'd expect that you don't care about the return value in the new  code and your update tracking will be using the new flow?